server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -586,7 +586,8 @@ public void deleteIndexDirectoryUnderLock(Index index, IndexSettings indexSettin
      * @param lockTimeoutMS how long to wait for acquiring the indices shard locks
      * @return the {@link ShardLock} instances for this index.
      */
-    public List<ShardLock> lockAllForIndex(Index index, IndexSettings settings, long lockTimeoutMS) throws ShardLockObtainFailedException {
+    public List<ShardLock> lockAllForIndex(Index index, IndexSettings settings, long lockTimeoutMS,
+                                           final String lockDetails) throws ShardLockObtainFailedException {
```

Comment:
I think we use the `final` modifier consistently? It looks odd that none of the parameters is declared `final` but the new one? (I will just mention it once here but this applies to all changed method signatures). 

Also, from an API perspective `lockDetails` is mandatory but the timeout is more like an optional thing to me (`#shardLock()` below has even an overloaded version where we don't need to specify a timeout. IMHO mandatory parameters should appear before optional ones so I think we should change the signature to:

```java
public List<ShardLock> lockAllForIndex(Index index, IndexSettings settings, String lockDetails, long lockTimeoutMS) throws ShardLockObtainFailedException
```

Reply:
> It looks odd that none of the parameters is declared `final` but the new one?

I like to mark parameters as final because it prevents accidental variable shadowing. I can change them all to final if you'd like (I just changed the one I added).

I'll change the ordering of the parameters.